### PR TITLE
CGCloud deploy docs

### PR DESCRIPTION
**Pull request by jpdna**

### CODE REVIEW

File: docs/source/40_deploying_ADAM.md [hunk 1]
```diff
@@ -1,57 +1,106 @@
...
+Note, the steps to register your ssh key and create the template boxes below need only be done once.
+```
+cgcloud register-key ~/.ssh/id_rsa.pub
+cgcloud create generic-ubuntu-trusty-box
+cgcloud create -IT spark-box
+```
+
+#### Launch a cluster
+
+Spin up a Spark cluster with one master and two slave nodes with the command:
```
Comment (fnothaft) [inline docs/source/40_deploying_ADAM.md]: Prefer leader/worker to master/slave.

Also, I would note in the documents that you're setting up a cluster where the workers are `m3.large`. Somewhat obvious, I concede, but it's useful to note that you can set a different leader node type. Also, doesn't this command need you to provide a cluster name?

File: docs/source/40_deploying_ADAM.md [hunk 2]
```diff
@@ -1,57 +1,106 @@
...
+CGCloud provides an automated means to create a cluster on EC2 for use with ADAM.
 
-```bash
-export AWS_ACCESS_KEY_ID='?????'
-export AWS_SECRET_ACCESS_KEY='?????'
-export MY_KEYPAIR="?????"      # your keypair in us-east
-export MY_KEYFILE="?????.pem"
-export MY_CLUSTER_NAME="adam_cluster"
-export MY_CLUSTER_SIZE=10
+[CGCloud](https://github.com/BD2KGenomics/cgcloud) lets you automate the creation, management and provisioning of VMs and clusters of VMs in Amazon EC2.
```
Comment (fnothaft) [inline docs/source/40_deploying_ADAM.md]: Can you wrap lines at 80 characters throughout?

File: docs/source/40_deploying_ADAM.md [hunk 3]
```diff
@@ -1,57 +1,106 @@
...
+Spin up a Spark cluster with one master and two slave nodes with the command:
+```
+cgcloud create-cluster spark -s 2 -t m3.large
+```
+
+Once running, you can ssh to `spark-master` with the command:
+```
+cgcloud ssh spark-master
+```
+
```
Comment (fnothaft) [inline docs/source/40_deploying_ADAM.md]: Nit: extra whitespace.

File: docs/source/40_deploying_ADAM.md [hunk 4]
```diff
@@ -1,57 +1,106 @@
...
 ```
 
-If you want to use spot pricing, add `--spot-price` as an option to
-`spark_ec2_launch` (below) and `export MY_SPOT_PRICE=1.399`.
+#### Install ADAM
 
-Export the path to your `spark-ec2` script,
+To use the ADAM application on top of Spark, we need to download and install ADAM on `spark-master`
+From the command line on `spark-master` download a release from:
+https://github.com/bigdatagenomics/adam/releases
```
Comment (fnothaft) [inline docs/source/40_deploying_ADAM.md]: Nit: missing period at EOL.

File: docs/source/40_deploying_ADAM.md [hunk 5]
```diff
@@ -1,57 +1,106 @@
...
-alias spark_ec2_launch="$SPARK_EC2_SCRIPT -k $MY_KEYPAIR \
--i $MY_KEYFILE -s $MY_CLUSTER_SIZE --zone us-east-1c \
---instance-type=$MY_INSTANCE_TYPE launch $MY_CLUSTER_NAME"
-alias spark_ec2_stop="$SPARK_EC2_SCRIPT stop $MY_CLUSTER_NAME"
-alias spark_ec2_start="$SPARK_EC2_SCRIPT -i $MY_KEYFILE start $MY_CLUSTER_NAME"
-alias spark_ec2_destroy="$SPARK_EC2_SCRIPT destroy $MY_CLUSTER_NAME"
-alias spark_ec2_login="$SPARK_EC2_SCRIPT -k $MY_KEYPAIR -i $MY_KEYFILE login $MY_CLUSTER_NAME"
+The typical flow of data to and from your ADAM application on EC2 will be:
+- Upload data to AWS S3
+- Use Conductor (described below) or otherwise transfer from S3 to the HDFS on your cluster
```
Comment (fnothaft) [inline docs/source/40_deploying_ADAM.md]: Can you add an anchor link `{#conductor}` in the section where conductor is described, and link from here `(described below)` -> `[(described below)](#conductor)`. This'll make navigation a bit easier.

File: docs/source/40_deploying_ADAM.md [hunk 6]
```diff
@@ -1,57 +1,106 @@
...
+From ADAM shell, or as parameter to ADAM submit, you would refer HDFS URLs  such as:
+```
+adam-submit transform hdfs://spark-master/work_dir/sample1.bam \
+                      hdfs://spark-master/work_dir/sample1.adam
+```
+
+To transfer large amounts of data back and forth from S3, we suggest using [Conductor](https://github.com/BD2KGenomics/conductor).
+
+Its also possible to directly use AWS S3 as a distributed file system, but with some loss of performance.
+( example to be added )
```
Comment (fnothaft) [inline docs/source/40_deploying_ADAM.md]: Nit: I might drop the `example to be added` bit and remove the paragraph break between this paragraph and the conductor paragraph.

**Comment by jpdna** [comment]
ready for further review or merge

File: docs/source/40_deploying_ADAM.md [hunk 7]
```diff
@@ -1,57 +1,136 @@
...
-```bash
-alias spark_ec2_launch="$SPARK_EC2_SCRIPT -k $MY_KEYPAIR \
--i $MY_KEYFILE -s $MY_CLUSTER_SIZE --zone us-east-1c \
---instance-type=$MY_INSTANCE_TYPE launch $MY_CLUSTER_NAME"
-alias spark_ec2_stop="$SPARK_EC2_SCRIPT stop $MY_CLUSTER_NAME"
-alias spark_ec2_start="$SPARK_EC2_SCRIPT -i $MY_KEYFILE start $MY_CLUSTER_NAME"
-alias spark_ec2_destroy="$SPARK_EC2_SCRIPT destroy $MY_CLUSTER_NAME"
-alias spark_ec2_login="$SPARK_EC2_SCRIPT -k $MY_KEYPAIR -i $MY_KEYFILE login $MY_CLUSTER_NAME"
+Spin up a Spark cluster named `cluster1` with one leader and two workers nodes 
+of instance type `m3.large`with the command:
```
Comment (fnothaft) [inline docs/source/40_deploying_ADAM.md]: Space between words in `m3.large`with

File: docs/source/40_deploying_ADAM.md [hunk 8]
```diff
@@ -1,57 +1,136 @@
...
+by starting a spark-shell.
+```
+spark-shell
+exit()
+```
+
+#### Install ADAM
+
+To use the ADAM application on top of Spark, we need to download and install 
+ADAM on `spark-master`
```
Comment (fnothaft) [inline docs/source/40_deploying_ADAM.md]: period at EOL

File: docs/source/40_deploying_ADAM.md [hunk 9]
```diff
@@ -1,57 +1,136 @@
...
+From the command line on `spark-master` download a release from:
+https://github.com/bigdatagenomics/adam/releases
```
Comment (fnothaft) [inline docs/source/40_deploying_ADAM.md]: Punctuation at EOL? Maybe remove paragraph break.

File: docs/source/40_deploying_ADAM.md [hunk 10]
```diff
@@ -1,57 +1,136 @@
...
+
+As of this writing, CGCloud supports Spark 1.6.2, not Spark 2.x, so download
+the Spark 1.x Scala2.10 release:
+```
+wget https://repo1.maven.org/maven2/org/bdgenomics/adam/\
```
Comment (fnothaft) [inline docs/source/40_deploying_ADAM.md]: I would remove the `\`ed linebreak here.

**Comment by jpdna** [comment]
ready again for more review or merge

**Comment by fnothaft** [comment]
Merged! Thanks @jpdna!

**Pull request merged**